### PR TITLE
Add BaseMeshOutline inspector

### PR DESCRIPTION
**Pull request by Proton-V**
## Changes
- Fixes: #10657 

## Overview
Added simple inspector helper for BaseMeshOutline abstract class, 
used in MeshOutline && MeshOutlineHierarchy, (mb in other custom classes)

## Includes
- check current material settings
- force update material settings to default outline mat settings
- add new correct material

## Added
- Inspector for BaseMeshOutline abstract class

### CODE REVIEW

**Review by keveleigh — COMMENTED** [review]
Thanks for this!

@Cameron-Micka might want to take a look as well

File: Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    /// Used for create or fix outline material
+    /// </summary>
+    [CustomEditor(typeof(BaseMeshOutline), true)]
+    public class BaseMeshOutlineInspector : UnityEditor.Editor
+    {
+        private BaseMeshOutline instance;
+        private SerializedProperty m_Script;
+        private SerializedProperty outlineMaterial;
+
+        private readonly string standardShaderName = "Mixed Reality Toolkit/Standard";
```
[the remark this answers is not in the document]
Comment (Proton-V) [reply Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs]: Oh, thanks

File: Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                            return !material.IsKeywordEnabled(x.Key);
+                        }
+                    default:
+                        // Default return value
+                        return false;
+                }
+            });
+        }
+
+        private bool IsCorrectShader(Material material)
```
Comment (Cameron-Micka) [inline Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs:146]: nit: You could replace this with [IsUsingMrtkStandardShader()](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/70d5731d959388a199143008d1c8ce7c4c4c577d/Assets/MRTK/Core/Utilities/StandardShader/StandardShaderUtility.cs#L46).

File: Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            ForceUpdateToDefaultOutlineMaterial(ref material);
+            AssetDatabase.CreateAsset(material, $"Assets/{Selection.activeGameObject.name}Mat.mat");
+            return material;
+        }
+
+        private void ForceUpdateToDefaultOutlineMaterial(ref Material material)
+        {
+            if (!IsCorrectShader(material))
+            {
+                material.shader = Shader.Find(MrtkStandardShaderName);
```
Comment (Cameron-Micka) [inline Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs:89]: nit: You could use [this ](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/70d5731d959388a199143008d1c8ce7c4c4c577d/Assets/MRTK/Core/Utilities/StandardShader/StandardShaderUtility.cs#L21)in lieu of Shader.Find.

**Review by Cameron-Micka — APPROVED** [review]
This is amazing, love it! I'll open up a task to port this to MRTK3 as well.

File: Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs (hunk 2 above)
Comment (Proton-V) [reply Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs]: Thank you!

File: Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs (hunk 3 above)
Comment (Proton-V) [reply Assets/MRTK/SDK/Editor/Inspectors/Utilities/MeshOutline/BaseMeshOutlineInspector.cs]: Nice!

**Comment by keveleigh** [comment]
Thanks for your contribution!

**Pull request merged**